packages/web3-utils/src/big_decimal.ts

```diff
@@ -0,0 +1,51 @@
+export class BigDecimal {
+	private readonly bigint: bigint;
+	private readonly decimals: number;
+
+	public constructor(value: bigint | string, decimals?: number) {
+		if (typeof value === 'string' && decimals) {
+			throw new Error('Can not specify decimal with string value');
+		}
+
+		if (typeof value === 'string') {
+			try {
+				const [integer, fraction] = String(value).split('.').concat('');
+				this.decimals = decimals ?? fraction.length;
+				this.bigint = BigInt(`${integer}${fraction.padEnd(this.decimals, '0')}`);
+			} catch {
+				throw new Error(`Invalid value given "${value}". Error: not a valid integer.`);
```

Comment:
We should avoid mixing inline errors with the use of `errors.ts`, no?

Reply:
Also this error doesn't make sense since we're splitting on the `.` here:

```typescript
const [integer, fraction] = String(value).split('.').concat('');
```

I think the error message should say `...not a valid number.` as opposed to `integer`